src/btree/bt_cursor.c

```diff
@@ -1155,8 +1156,20 @@ __wt_btcur_remove(WT_CURSOR_BTREE *cbt, bool positioned)
 
             WT_WITH_UPDATE_VALUE_SKIP_BUF(ret = __wt_cursor_valid(cbt, NULL, cbt->recno, &valid));
             WT_ERR(ret);
-            if (!valid)
+            if (!valid && btree->type != BTREE_COL_FIX)
                 WT_ERR(WT_NOTFOUND);
+            else if (!valid) {
+                /*
+                 * To preserve the illusion that deleted values are 0 and that we can therefore
+                 * delete them, without violating the system restriction against consecutive
+                 * tombstones, generate a dummy value for the new tombstone to delete. Make the
+                 * dummy value zero for good measure.
+                 */
+                cbt->v = 0;
+                flcs_zero.data = &cbt->v;
+                flcs_zero.size = 1;
+                WT_ERR(__cursor_col_modify(cbt, &flcs_zero, WT_UPDATE_STANDARD));
+            }
```

Comment:
Can't we just do this with `static const WT_ITEM flcs_zero = { "", 1, NULL, 0, 0};`?

Reply:
Wasn't sure if that was allowable. Anyway, yes (though it apparently can't be const); fixed in [0e146ad](https://github.com/wiredtiger/wiredtiger/pull/7746/commits/0e146ad09c260a84d4560fbef72def4cd789958c).

Reply:
Ah, it can be const, seems that __cursor_col_modify throws away __wt_col_modify's const. Fixing that...

Reply:
Done in [aa38f2e](https://github.com/wiredtiger/wiredtiger/pull/7746/commits/aa38f2e8bd66dbd8f4341021dc264ca0cf2c4456).